information for modern samples added

93Boy opened the pull request:
<!--
Hello there!

Thanks for either
i. preparing a new package for submission to the community archive or
ii. improving one or multiple packages.

Both tasks are described in the submission guide here: https://www.poseidon-adna.org/#/archive_submission_guide

You can now open this Pull Request and then work through the relevant checklists below.
-->

## PR Checklist for modifying one or multiple existing packages

- [x] The changes maintain the structural integrity of the affected packages.
- [x] The checksums of the modified files in the respective `POSEIDON.yml` files were adjusted properly.
- [x] Every file in the submission is correctly referenced in the relevant `POSEIDON.yml` files and there are no additional, supplementary files in the submission that are not documented there.

***

- [x] The `packageVersion` numbers of the affected packages were increased in their `POSEIDON.yml` files.
- [x] The changes were documented in the respective `CHANGELOG` files. If no `CHANGELOG` files existed previously it was added here.
- [x] The `lastModified` fields of the affected `POSEIDON.yml` files were updated.
- [x] The `contributor` fields were updated with `name`, `email` and `orcid` of the relevant, new contributors.

***

- [x] All affected packages pass a validation with `trident validate --fullGeno`.

***

- [x] Large genotype data files are properly tracked with Git LFS and not directly pushed to the repository. For an instruction on how to set up Git LFS please look [here](https://www.poseidon-adna.org/#/archive_submission_guide?id=submitting-the-package). If you accidentally pushed the files the wrong way you can fix it with `git lfs migrate import --no-rewrite path/to/file.bed` (see [here](https://github.com/git-lfs/git-lfs/blob/main/docs/man/git-lfs-migrate.adoc#import-without-rewriting-history)).

nevrome commented:
Cool that you started to add this info, @93Boy! Could you go through the checklist above and mark what you have done?

93Boy commented:
Sure , I have a few more fields to complete in KilincSciAdv2021. I will check the package information afterwards

stschiff commented:
quotations!

stschiff commented:
Also there are encoding problems. Make sure the text export is UTF8 (e.g. "Tai‚ÄìKadai")

stschiff commented:
Can we get an update on this one, @93Boy, please?

stschiff commented:
Please go through the checklist, and in particular update the versions. We also have seen backticks in locations and haplogroups in the Janno-file. Please clean them up @93Boy.

stschiff commented:
Please update the Changelog and the version in the YAML. Also there are still single quotes (`'''''''`) in the Wang_EastAsia janno file. Please fix.

93Boy commented:
I have made the informed changes. Please check and merge if everything is good

stschiff commented:
Hi @93Boy. Thanks, but please remember to always check the "Files Changes" tab. You can easily see that the Janno has quotations. Please remove them.

93Boy commented:
I am extremely sorry for this quotation issue. It's a default setting in Libre Office which I can't bypass. It will simply change the structure of a file, even if I open it to check.

stschiff reviewed (approved):
OK, looks like this is finally good to merge. @AyGhal, @nevrome do you agree?

AyGhal reviewed (commented):
Looks good to me. Thank you! 
My only question is do we need to update `lastModified:` for 2021_Wang_EastAsia ?

stschiff commented:
Excellent catch, @AyGhal. I have just done that inline. Now it should be good to merge.

nevrome commented:
I see many new warnings of the form:

```
[Warning] [2024-03-15 22:31:08] Issue in data/2021_Kilinc_northeastAsia/Kilinc_northeastAsia.janno in line 2 (Poseidon_ID: bla001): Date_Type is "C14", but either Date_C14_Uncal_BP or Date_C14_Uncal_BP_Err are empty
```

Note that this check was only added to trident in v1.4.1.0. It should be an error, but we have some legacy packages with this issue, so we decided to only make it a warning. We should absolutely avoid it for new packages.

@93Boy: You seem to get this error, because `Date_Type` is set to `C14` in `2021_Kilinc_northeastAsia`, but `Date_C14_Uncal_BP` and `Date_C14_Uncal_BP_Err` are empty. We only seem to have `contextual` dates in this .janno file. These columns seem actually to be completely empty, so they might as well be removed from the .janno file.

@stschiff and @AyGhal: I made it a habit to search the GitHub validation log for the keyword `warning` in my reviews. Maybe we should also add this to the submission checklist.

stschiff commented:
OK, I will run the validator and fix the outstanding issues quickly

stschiff commented:
OK, I think indeed the Kilinc package lists C14 dates but no IDs or uncalibrated dates. I will take a look at the paper myself and perhaps add the dates.

stschiff commented:
OK, so I've checked the paper (https://www.science.org/doi/10.1126/sciadv.abc4587) and here attach the supplementary Tables that list the C14 dates. @93Boy can you please go through these and add the C14 IDs and uncalibrated dates to the Janno file? Alternatively, if you feel that would take too much time for now, we can also merge this PR for now and then fix this separately. But I think it could be a relatively quick task of about one hour to add all of them.

[abc4587_data_file_s1.xlsx](https://github.com/poseidon-framework/community-archive/files/14747022/abc4587_data_file_s1.xlsx)

93Boy commented:
I have updated Date_C14_Labnr and Date_C14_Uncal_BP based on the supplementary materials, but afterward, there are several consistency issues in lines 2 to 6 in the Dates columns. But as I see these rows are according to PCA standards. Can you kindly check this?

stschiff commented:
You just need to add the C14 Error, which is reported in the column named "SD" in the Supplementary Table

93Boy commented:
Changes updated, now it looks fine on my end

nevrome commented:
Upon closer inspection I see the following issues:

- **2021_Kilinc_northeastAsia**
    - The values that were added now for the `Collection_ID` variable (e.g. *Russia_Blagovechensk_IA.SG* or *Russia_AngaraRiver_Medieval.SG*) seem to be actually the intended `Group_Name`. The current `Group_Name`s are just the `Poseidon_ID`s. Adding these values as the `Group_Name` would require also to adjust the .fam file.
    - @93Boy's editor once more added some instances of excessive single quotes *'''''''* in the `Site` and `MT_Haplogroup` columns.
    - The content of the `Site` column seems to fit better into the `Location` column.
- **2021_Wang_EastAsia**
    - The `Location` column often includes the `Y_Haplogroup` value.
    - There seems to be an encoding issue in the `Location` column: *Tai-Kadai* is often written as *Tai‚ÄìKadai*
    - The `Capture_Type` *1240K* does not fit for the modern data, right?

We could split the work to fix them :+1:

93Boy commented:
@nevrome I have removed the typos caused by my editor and changed group names. I have changed .fam files accordingly but trident-rectify throws the following error. `Cross-file consistency issue in package 2021_Kilinc_northeastAsia: Individual GroupID mismatch between genotype data (left) and .janno files (right). Note that this could be due to a wrong Plink file population-name encoding (see the --inPlinkPopName option). (Russia_AginBuryat_M.SG = Russia_ArgunRiver_M.SG)`

stschiff commented:
OK, so the group names between the janno and fam files did not match up. Upon close inspection I found a mismatch on line 7, where one says `Russia_AginBuryat_M.SG` and the other `Russia_ArgunRiver_M.SG`. I will find out which is right

93Boy commented:
@stschiff Regarding the `2021_Wang_EastAsia` , when I took over this package there were 383 entries that were incomplete starting from the poseidon_ID `GN01` and ending from `Dong007`. They are the modern samples according to my knowledge. Later on, I extracted 191 ancient entries from AADR v50. They are complete and in the correct format. This issue has been there even before the commit you mentioned above.

stschiff commented:
@93Boy I've updated my comment. The history how this came about isn't too important. The issues are relatively clear. Can you work on fixing them, please, and let us know where you couldn't? Thanks.

93Boy commented:
I have made the following recommended changes to 2021_Wang_EastAsia Janno,

1. Y_haplo group information was extracted from the Location column for all the 383 modern samples and updated.
2. After examining the Group names, they are more likely locations. Therefore I swapped group names and location data. (Please let me know , if Im wrong I can easily change it)

I have thoroughly reviewed supplementary documents, but I could not find the information for 383 modern samples. I appreciate it if you could guide me about this. Herewith I have attached the supplementary documents.
[41586_2021_3336_MOESM3_ESM (2).zip](https://github.com/poseidon-framework/community-archive/files/15180595/41586_2021_3336_MOESM3_ESM.2.zip)

stschiff commented:
Notes from Meeting on May 3rd:

For Wang et al. 2021:

- please remove quotes
- What you currently added as "location" for the modern samples, are in fact languages. Please remove them. The original group names from the Master branch were correct, please put them back in for the modern samples. They are now in "location", where they can also stay. So you can - for the modern samples only - have same entries in group and location.
- Probably fine to leave "Collection_ID" as "n/a" for the modern samples.
- For the ancient samples, the "Collection_ID" is redundantly copied from the Poseidon_ID. Please use column "Skeletal Code" from Online Table 1.
- For ancient samples: Please put back the location info from the master branch. It got lost.

93Boy commented:
Wang et al. 2021: 
I have done the requested changes. Furthermore I have notices several (30+) poseidon_IDs were actually Library_ids according to the supplementary materials and I could have extracted a proper ID key from those supplementary materials. But now trident rectify can not recognize these new poseidon_Ids. May I change fam files too?

93Boy commented:
[suspicious_IDs.csv](https://github.com/poseidon-framework/community-archive/files/15340022/suspicious_IDs.csv)
I have done a small analysis regarding the problematic poseidon_IDs of Wang et.al 2021 and finding are given below, 

- There are 28 poseidon_IDs which are most likely library codes. 
- I was able to match a valid poseidon_IDs for all of these entries from the old collection_IDs which we replaced with skeletal codes under the commit "e0b4031"
- AADR v54.1 also contain 18 of these suspicious_IDs, they also follow the format of old collection_IDs

stschiff commented:
Notes from May 17th:

Thanks for catching these. As discussed. Please replace library names like "S15162.Y1.E1.L1" with the Master ID in Supplementary Table 2, in this case it would be "I15162", wherever this is possible.

stschiff commented:
Update, @93Boy ?

93Boy commented:
I completed all the requested changes during the last review. I think now this is ready for the second review

nevrome commented:
2021_Kilinc_northeastAsia looks good to me now.

nevrome commented:
2021_Wang_EastAsia also looks good now, imho.

stschiff reviewed (approved):
Super. Good to go from my end.

nevrome commented:
After an epic seven month long quest this PR got finally merged :rocket:
